systemvm/patches/debian/config/opt/cloud/bin/cs/CsAddress.py

```diff
@@ -452,9 +452,9 @@ def fw_vpcrouter(self):
             self.fw.append(
                 ["filter", "", "-A INPUT -i %s -p udp -m udp --dport 67 -j ACCEPT" % self.dev])
             self.fw.append(
-                ["filter", "", "-A INPUT -i %s -p udp -m udp --dport 53 -j ACCEPT" % self.dev])
+                ["filter", "", "-A INPUT -i %s -p udp -m udp --dport 53 -s %s -j ACCEPT" % (self.dev, self.address['network'])])
             self.fw.append(
-                ["filter", "", "-A INPUT -i %s -p tcp -m tcp --dport 53 -j ACCEPT" % self.dev])
+                ["filter", "", "-A INPUT -i %s -p tcp -m tcp --dport 53 -s %s -j ACCEPT" % (self.dev, self.address['network'])])
```

Comment:
Minor nit: extract the value of ``self.address['network']`` to a local variable?  Not only will change ensure consistency of the value across the four rules, but it will make the code more literate.